File: app/src/main/java/com/example/ui/components/CurioLogo.kt
package com.example.ui.components

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.layout.size
import androidx.compose.material3.MaterialTheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.geometry.Size
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.Path
import androidx.compose.ui.graphics.PathFillType
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import com.example.ui.theme.BookmarkTheme

/**
 * Curio brand mark: a bookmark ribbon with an AI "spark" knocked out of it.
 *
 * Drawn with the SAME 108x108 geometry as the launcher icon (res/drawable/
 * ic_launcher_foreground.xml), so the in-app logo and the home-screen icon are
 * pixel-identical in shape. The spark is a genuine [PathFillType.EvenOdd] hole,
 * so whatever sits behind the mark shows through it.
 *
 * Theme-aware: [tint] defaults to [MaterialTheme.colorScheme] onSurface, so the
 * mark flips white-on-dark / black-on-light (and follows Material You dynamic
 * color) automatically. Used bare in headers (e.g. LoginScreen glass circle).
 */
@Composable
fun CurioLogoMark(
    modifier: Modifier = Modifier,
    tint: Color = MaterialTheme.colorScheme.onSurface,
    paddingFraction: Float = 0.04f,
) {
    Canvas(modifier = modifier) {
        drawPath(curioMarkPath(size, paddingFraction), color = tint)
    }
}

/**
 * Builds the bookmark + spark path, scaled to fit [size] (aspect-preserved,
 * centered) with [paddingFraction] breathing room on the limiting axis.
 * Coordinates are authored in the bookmark's 108-viewport bounding box
 * (x:37..71, y:30..80) so they line up 1:1 with the vector drawable.
 */
private fun curioMarkPath(size: Size, paddingFraction: Float): Path {
    val originX = 37f
    val originY = 30f
    val markW = 34f // 71 - 37
    val markH = 50f // 80 - 30

    val scale = minOf(
        size.width * (1f - 2f * paddingFraction) / markW,
        size.height * (1f - 2f * paddingFraction) / markH,
    )
    val offsetX = (size.width - markW * scale) / 2f
    val offsetY = (size.height - markH * scale) / 2f

    fun x(v: Float) = offsetX + (v - originX) * scale
    fun y(v: Float) = offsetY + (v - originY) * scale

    return Path().apply {
        fillType = PathFillType.EvenOdd

        // Bookmark ribbon with rounded top corners and a V-notch tail.
        moveTo(x(46f), y(30f))
        lineTo(x(62f), y(30f))
        quadraticTo(x(71f), y(30f), x(71f), y(39f))
        lineTo(x(71f), y(80f))
        lineTo(x(54f), y(66f))
        lineTo(x(37f), y(80f))
        lineTo(x(37f), y(39f))
        quadraticTo(x(37f), y(30f), x(46f), y(30f))
        close()

        // Four-point AI spark — knocked out of the ribbon via EvenOdd.
        moveTo(x(54f), y(35f))
        quadraticTo(x(56.40f), y(45.60f), x(67f), y(48f))
        quadraticTo(x(56.40f), y(50.40f), x(54f), y(61f))
        quadraticTo(x(51.60f), y(50.40f), x(41f), y(48f))
        quadraticTo(x(51.60f), y(45.60f), x(54f), y(35f))
        close()
    }
}

@Preview(name = "Bare mark — light", showBackground = true, backgroundColor = 0xFFFEF7FF)
@Preview(name = "Bare mark — dark", showBackground = true, backgroundColor = 0xFF141218)
@Composable
private fun CurioLogoMarkPreview() {
    BookmarkTheme(dynamicColor = false) {
        CurioLogoMark(modifier = Modifier.size(72.dp))
    }
}

File: ios/Curio/Curio/Components/CurioLogo.swift
//
//  CurioLogo.swift
//  Curio
//
//  Ports: app/src/main/java/com/example/ui/components/CurioLogo.kt
//         (CurioLogoMark, the private `curioMarkPath` geometry).
//
//  CONVENTIONS §8 (Theme/Liquid-Glass) + DESIGN: the brand mark is the SAME 108x108
//  geometry as the launcher icon (res/drawable/ic_launcher_foreground.xml) so the in-app
//  logo and the home-screen icon are pixel-identical in shape. The spark is a genuine
//  even-odd hole (`.evenOdd` fill), so whatever sits behind the mark shows through it.
//
//  Compose `Canvas { drawPath(curioMarkPath(size, paddingFraction)) }` becomes a
//  `Shape` (`CurioMarkShape`) so the path math is reused verbatim and the even-odd
//  spark cutout is preserved (`Path(eoFill: true)` → SwiftUI fills with `.evenOdd`).
//  Theme-aware tint defaults to `onSurface` (read via `\.curioColors`),
//  mirroring the Material `MaterialTheme.colorScheme` defaults in Kotlin.
//

import SwiftUI

// MARK: - CurioMarkShape

/// The bookmark + spark brand path, authored in the bookmark's 108-viewport bounding box
/// (x:37..71, y:30..80) so it lines up 1:1 with the vector drawable. Scaled to fit the
/// supplied `rect` (aspect-preserved, centered) with `paddingFraction` breathing room on the
/// limiting axis. The spark is a four-point star knocked out of the ribbon via even-odd fill.
///
/// Verbatim port of the private `curioMarkPath(size:paddingFraction:)` in CurioLogo.kt; the
/// Compose `quadraticTo(cx, cy, x, y)` maps to `path.addQuadCurve(to:control:)`.
struct CurioMarkShape: Shape {
    /// Breathing-room margin on the limiting axis, as a fraction of that axis. Compose default
    /// for the bare mark is `0.04`.
    var paddingFraction: CGFloat = 0.04

    func path(in rect: CGRect) -> Path {
        let originX: CGFloat = 37
        let originY: CGFloat = 30
        let markW: CGFloat = 34 // 71 - 37
        let markH: CGFloat = 50 // 80 - 30

        let scale = min(
            rect.width * (1 - 2 * paddingFraction) / markW,
            rect.height * (1 - 2 * paddingFraction) / markH
        )
        let offsetX = (rect.width - markW * scale) / 2
        let offsetY = (rect.height - markH * scale) / 2

        // Account for the rect origin (SwiftUI shapes may be offset from 0,0).
        func x(_ v: CGFloat) -> CGFloat { rect.minX + offsetX + (v - originX) * scale }
        func y(_ v: CGFloat) -> CGFloat { rect.minY + offsetY + (v - originY) * scale }

        var path = Path()

        // Bookmark ribbon with rounded top corners and a V-notch tail.
        path.move(to: CGPoint(x: x(46), y: y(30)))
        path.addLine(to: CGPoint(x: x(62), y: y(30)))
        path.addQuadCurve(to: CGPoint(x: x(71), y: y(39)), control: CGPoint(x: x(71), y: y(30)))
        path.addLine(to: CGPoint(x: x(71), y: y(80)))
        path.addLine(to: CGPoint(x: x(54), y: y(66)))
        path.addLine(to: CGPoint(x: x(37), y: y(80)))
        path.addLine(to: CGPoint(x: x(37), y: y(39)))
        path.addQuadCurve(to: CGPoint(x: x(46), y: y(30)), control: CGPoint(x: x(37), y: y(30)))
        path.closeSubpath()

        // Four-point AI spark — knocked out of the ribbon via even-odd fill. Verbatim port of
        // the four Compose `quadraticTo(cx, cy, x, y)` calls (control then end point).
        path.move(to: CGPoint(x: x(54), y: y(35)))
        path.addQuadCurve(to: CGPoint(x: x(67), y: y(48)), control: CGPoint(x: x(56.40), y: y(45.60)))
        path.addQuadCurve(to: CGPoint(x: x(54), y: y(61)), control: CGPoint(x: x(56.40), y: y(50.40)))
        path.addQuadCurve(to: CGPoint(x: x(41), y: y(48)), control: CGPoint(x: x(51.60), y: y(50.40)))
        path.addQuadCurve(to: CGPoint(x: x(54), y: y(35)), control: CGPoint(x: x(51.60), y: y(45.60)))
        path.closeSubpath()

        return path
    }
}

// MARK: - CurioLogoMark

/// Curio brand mark: a bookmark ribbon with an AI "spark" knocked out of it.
///
/// Drawn with the same 108x108 geometry as the launcher icon, so the in-app logo and the
/// home-screen icon are pixel-identical in shape. The spark is a genuine even-odd hole, so
/// whatever sits behind the mark shows through it. `tint` defaults to the Cosmic `onSurface`
/// (mirrors Compose `MaterialTheme.colorScheme.onSurface`), flipping white-on-dark /
/// dark-on-light automatically.
struct CurioLogoMark: View {
    var tint: Color? = nil
    var paddingFraction: CGFloat = 0.04

    @Environment(\.curioColors) private var colors

    var body: some View {
        CurioMarkShape(paddingFraction: paddingFraction)
            .fill(tint ?? colors.onSurface, style: FillStyle(eoFill: true))
            .accessibilityHidden(true)
    }
}

// MARK: - Previews

#Preview("Bare mark — dark") {
    CurioLogoMark()
        .frame(width: 72, height: 72)
        .padding()
        .background(Color(argb: 0xFF141218))
        .environment(\.curioColors, .dark)
}

#Preview("Bare mark — light") {
    CurioLogoMark()
        .frame(width: 72, height: 72)
        .padding()
        .background(Color(argb: 0xFFFEF7FF))
        .environment(\.curioColors, .light)
}
